src/App.js

```diff
@@ -1,13 +1,97 @@
 import React from 'react';
+import TodoForm from './components/TodoComponents/TodoForm'
+import TodoList from './components/TodoComponents/TodoList'
+import './components/TodoComponents/Todo.css';
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.testRef = React.createRef();
+    this.state = {
+      currentInputValue: '',
+      todos : []
+      };
+      window.onbeforeunload = this.saveData;
+  }
+  
+  componentDidMount(){
+    const data = JSON.parse(localStorage.getItem('data'));
+
+    if(data !== null) {
+      this.setState({
+        todos: JSON.parse(localStorage.getItem('data'))
+      })
+    }
+  };
+
+  saveData = () => {
+    localStorage.setItem('data', JSON.stringify(this.state.todos))
+  };
+
+  handleInput = (event) => {
+    this.setState({
+      currentInputValue: event.target.value
+    });
+  }
+
+  handleSubmit = (event) => {
+
+    if (this.state.currentInputValue === '') {
```

Comment:
Good job checking for empty todos, how about the case of multiple spaces for an "empty" todo?